### ISSUE TITLE
Strange (redundand ?) symbols in *.bin

### ISSUE THREAD

**Issue by wicked-code** [espressif/ESP8266_RTOS_SDK#141]
There are many <finame>.c conts in result bin file. Some of them occur several times in a row (one by one). For example "dhcpserver.c" occurs 14 times in eagle.irom0text.bin. It looks strange. Yes, it takes just 224 bytes, but this is only one example. Probably it's posible to free several kilobytes...

**Comment by 27-301** [comment]
Yes, it's a great discovery. I think it's caused by the os_printf macro:
```c
#define os_printf(fmt, ...) do {    \
        static const char flash_str[] ICACHE_RODATA_ATTR STORE_ATTR = fmt;  \
        printf(flash_str, ##__VA_ARGS__);   \
    } while(0)
#endif
```
Since currently SDK supports to put .rodata directly into flash, os_printf can be replaced by printf in most cases. This can indeedly reduce the final image size.
